data/shared/src/main/scala/sigma/data/CSigmaDslBuilder.scala

```diff
@@ -200,6 +204,29 @@ class CSigmaDslBuilder extends SigmaDslBuilder { dsl =>
     val p = GroupElementSerializer.parse(r)
     this.GroupElement(p)
   }
+
+  def deserializeRaw[T](tpe: SType, bytes: Coll[Byte])(implicit cT: RType[T]): T = {
+    val res = cT.classTag match {
+      case ClassTag.Short => ByteBuffer.wrap(bytes.toArray).getShort
+      case ClassTag.Int => scorex.utils.Ints.fromByteArray(bytes.toArray)
+      case ClassTag.Long => byteArrayToLong(bytes)
+      case sigma.data.BigIntClassTag => byteArrayToBigInt(bytes)
+    /*  case sigma.data.SigmaPropClassTag =>
+        ErgoTreeSerializer.DefaultSerializer.deserializeErgoTree(bytes.toArray).root match {
+          case Left(_) => throw new InvalidType(s"Cannot deserialize($bytes): unparsed tree provided")
+          case Right(prop) =>
+            if(prop.isInstanceOf[EvaluatedValue[_]]) {
+              prop.asInstanceOf[EvaluatedValue[_]].value
+            } else {
+              throw new InvalidType(s"Cannot deserialize($bytes): prop is not evaluated: $prop}")
+            }
+        } */
+      case _ =>
+        val reader = new SigmaByteReader(new VLQByteBufferReader(ByteBuffer.wrap(bytes.toArray)), new ConstantStore(), false)
+        DataSerializer.deserialize(tpe, reader)
```

Comment:
One problem with having special cases for Int, Long, BigInt is that their format will be different whey they are top level vs they are in a tuple or Coll, i.e. `serialize((intValue, longValue)) != serialize(intValue) ++ serialize(longValue)` and also `serialize(intValue) != serialize(Coll(intValue)).drop(1)`.
In other words, when the value inside a structure, it is suddenly has a different format.

I think it is better to just use DataSerializer, keep the implementation simple and generic, and forget about potential differences with other existing methods which return bytes of something.

Reply:
It seems you have not understood the proposal of mine expressed over Discord:

> 
> Investigating using DataSerializer, with it round-trip property with existing serializers (longtoByteArray, .propBytes etc) would be broken. Which would lead to following possible consequences:
> 
>     deserialization of those those types (Long, BigInt, SigmaProp) and similar (Short, Int) is corresponding to existing serializers, for other options DataSerializer is used
> 
> 
>     new "serialize: SAny -> SByteArray" option is introduced , with DataSerializer under the hood, but then there are two coexisting options, old one (longToByteArray/byteArrayToLong), and new one serialize/deserialize, for round-tripping, with the exception for .propBytes which can be deserialized via skipping first two bytes though, like " { val bytes = p1.propBytes; val ba = bytes.slice(2, bytes.size); val prop = Global.deserialize[SigmaProp](ba)}
> 
> 
> in case of serialize() (which is the better option I support), there is no need for header.bytes. avlTree.bytes, numeric.bytes (and the latter option would be confusing, as producing different bytes from *ToByteArray). Thus you can take serialize() instead of Numeric.toBytes